### ISSUE TITLE
MD053: False positive - reports used link references as unused

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#269]
## Summary

MD053 (link-image-reference-definitions) incorrectly reports link references as unused even when they are clearly used in the document.

## Example

From `ch00-00-introduction.md`:
```markdown
> Language][nsprust] available in print and ebook format from [No Starch
> Press][nsp].

[nsprust]: https://nostarch.com/rust-programming-language-3rd-edition
[nsp]: https://nostarch.com/
```

Both `[nsprust]` and `[nsp]` are used on lines 4 and 5, but we report:
```
ch00-00-introduction.md:7:1:warning: MD053/link-image-reference-definitions: Reference definition 'nsprust' is unused
ch00-00-introduction.md:8:1:warning: MD053/link-image-reference-definitions: Reference definition 'nsp' is unused
```

## Another Example

From `appendix-01-keywords.md`:
```markdown
Identifiers"][raw-identifiers]<!-- ignore --> section). _Identifiers_ are names

[raw-identifiers]: #raw-identifiers
```

The reference IS used but we report it as unused.

## Impact

On the Rust Book corpus:
- mdbook-lint: 107 violations
- rumdl: 5 violations  
- markdownlint: 5 violations

Most of our 107 violations are false positives.

## Root Cause

The rule is likely not correctly parsing reference-style links that use the `][ref]` syntax (where the link text and reference are separate), or not handling references inside blockquotes.

## Related

Issue #263 - Linter comparison
